From 702e4ff9f5fbfae4aecba09f8c735dc97bac1b59 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 22 Jan 2012 11:58:09 -0800
Subject: [PATCH 2/3] Cleanup shadow declarations

Box.c:265:10: warning: declaration shadows a local variable [-Wshadow]
            int i = bbw->composite.num_children;
                ^
Box.c:166:15: note: previous declaration is here
    Cardinal  i;
              ^
1 warning generated.

Form.c:809:9: warning: declaration shadows a local variable [-Wshadow]
        Widget w = *childP;
               ^
Form.c:793:24: note: previous declaration is here
XawFormDoLayout(Widget w,
                       ^
1 warning generated.

Viewport.c:627:24: warning: declaration shadows a local variable [-Wshadow]
                    if (!needshoriz) CheckHoriz();
                                     ^
Viewport.c:605:14: note: expanded from macro 'CheckHoriz'
                    Widget bar = w->viewport.horiz_bar;                 \
                           ^
Viewport.c:620:14: note: previous declaration is here
                    Widget bar = w->viewport.vert_bar;
                           ^
1 warning generated.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 src/Box.c      |    4 ++--
 src/Form.c     |    6 +++---
 src/Viewport.c |   20 ++++++++++----------
 3 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/src/Box.c b/src/Box.c
index ccd52fd..43be5dc 100644
--- a/src/Box.c
+++ b/src/Box.c
@@ -262,9 +262,9 @@ DoLayout(BoxWidget bbw, Dimension width, Dimension height,
 	if (bbw->composite.num_children == num_mapped_children)
 	    XMapSubwindows( XtDisplay((Widget)bbw), XtWindow((Widget)bbw) );
 	else {
-	    int i = bbw->composite.num_children;
+	    int j = bbw->composite.num_children;
 	    Widget *childP = bbw->composite.children;
-	    for (; i > 0; childP++, i--)
+	    for (; j > 0; childP++, j--)
 		if (XtIsRealized(*childP) && XtIsManaged(*childP) &&
 		    (*childP)->core.mapped_when_managed)
 		    XtMapWidget(*childP);
diff --git a/src/Form.c b/src/Form.c
index 49fceb6..f078270 100644
--- a/src/Form.c
+++ b/src/Form.c
@@ -790,7 +790,7 @@ PreferredGeometry(Widget widget, XtWidgetGeometry *request, XtWidgetGeometry *re
  */
 
 void
-XawFormDoLayout(Widget w,
+XawFormDoLayout(Widget _fw,
 #if NeedWidePrototypes
 		int doit)
 #else
@@ -798,11 +798,11 @@ XawFormDoLayout(Widget w,
 #endif
 {
     Widget *childP;
-    FormWidget fw = (FormWidget)w;
+    FormWidget fw = (FormWidget)_fw;
     int num_children = fw->composite.num_children;
     WidgetList children = fw->composite.children;
 
-    if ( ((fw->form.no_refigure = !doit) == TRUE) || !XtIsRealized(w) )
+    if ( ((fw->form.no_refigure = !doit) == TRUE) || !XtIsRealized(_fw) )
 	return;
 
     for (childP = children; childP - children < num_children; childP++) {
diff --git a/src/Viewport.c b/src/Viewport.c
index abb8961..a37a9a1 100644
--- a/src/Viewport.c
+++ b/src/Viewport.c
@@ -602,12 +602,12 @@ ComputeLayout(Widget widget, Boolean query, Boolean destroy_scrollbars)
 	    if (w->viewport.allowhoriz &&				\
 		    (int)preferred.width > clip_width + 2 * sw) {	\
 		if (!needshoriz) {					\
-		    Widget bar = w->viewport.horiz_bar;			\
+		    Widget horiz_bar = w->viewport.horiz_bar;		\
 		    needshoriz = True;					\
-		    if (bar == (Widget)NULL)				\
-			bar = CreateScrollbar(w, True);			\
-		    clip_height -= bar->core.height +			\
-				   bar->core.border_width + pad;	\
+		    if (horiz_bar == (Widget)NULL)			\
+			horiz_bar = CreateScrollbar(w, True);		\
+		    clip_height -= horiz_bar->core.height +		\
+				   horiz_bar->core.border_width + pad;	\
 		    if (clip_height < 1) clip_height = 1;		\
 		}							\
 		intended.width = preferred.width;			\
@@ -617,12 +617,12 @@ ComputeLayout(Widget widget, Boolean query, Boolean destroy_scrollbars)
 	    if (w->viewport.allowvert &&
 		    (int)preferred.height > clip_height + 2 * sw) {
 		if (!needsvert) {
-		    Widget bar = w->viewport.vert_bar;
+		    Widget vert_bar = w->viewport.vert_bar;
 		    needsvert = True;
-		    if (bar == (Widget)NULL)
-			bar = CreateScrollbar(w, False);
-		    clip_width -= bar->core.width +
-				  bar->core.border_width + pad;
+		    if (vert_bar == (Widget)NULL)
+			vert_bar = CreateScrollbar(w, False);
+		    clip_width -= vert_bar->core.width +
+				  vert_bar->core.border_width + pad;
 		    if (clip_width < 1) clip_width = 1;
 		    if (!needshoriz) CheckHoriz();
 		}
-- 
1.7.7.4

